packages/datagrid/src/datagrid.ts

```diff
@@ -6908,8 +6926,13 @@ namespace Private {
      * @param index - The index of column in the region.
      *
      * @param size - The target size of the section.
+     *               If null, then infer the size to fit.
      */
-    constructor(region: DataModel.ColumnRegion, index: number, size: number) {
+    constructor(
+      region: DataModel.ColumnRegion,
+      index: number,
+      size: number | null
```

Comment:
This is not very null-safe at the moment but will be by the time I convert this draft into a mergeable PR.